department-of-veterans-affairs/abd-vro
# MCP-1969 End-to-end test phase 1

## What was the problem?

There was no end to end test which was similar to what QA team performing.  With libraries changing frequently we need one.



There was no end to end test.

Associated tickets or Slack threads:

- [MCP-1969](https://amida.atlassian.net/browse/MCP-1969)

## How does this fix it?[^1]


This adds a new gradle task ":app:end2endTest". This task assumes the app docker-compose has already been called and all services are running locally. Based on gold files this test calls rest end points to generate health assessment and evidence pdf, and verifies the results.

Currently this test is intended to run only locally. Once this PR is cleared the test will be put in github workflows. That is the next step.

Notes

1) This depends on on two existing records on Lighthouse Health API sandbox. Future versions might use a mock api to avoid failures due to changes to those patients or filtering changes based on dates (ex: this test will fail in July since one of the blood pressure readings is in June).
2) Evidence PDF data fields are only sanity checked with counts. Future versions might use a more sophisticated parser to extract everything.
3) No negative or break tests yet. Those are for future.

## How to test this PR
- Make sure all the services are running ex "./gradlew resetloop".
- Run "./gradlew end2endTest" and make sure it passes. 
- You can change various things in the gold files to make sure it fails. For example change a bp value or remove a medication.

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
